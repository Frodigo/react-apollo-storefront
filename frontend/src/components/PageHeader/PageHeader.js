import React from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import HeaderActions from '../HeaderActions';

import Navigation from '../Navigation';
import PageLogo from '../PageLogo';
import QuickSearch from '../QuickSearch'

const PageHeader = () => {
    return (
        <header>
            <Container className="d-none d-lg-block">
                <Row className="d-flex align-items-center pb-3 pt-3">
                    <Col sm={3}>
                        <PageLogo/>
                    </Col>
                    <Col sm={6}>
                        <QuickSearch/>
                    </Col>
                    <Col sm={3} className="d-flex">
                        <HeaderActions/>
                    </Col>
                </Row>
            </Container>
            <Navigation/>
        </header>
    );
};


export default PageHeader;
