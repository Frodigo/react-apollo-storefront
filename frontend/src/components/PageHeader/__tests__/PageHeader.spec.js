import React from 'react';
import { render } from '@testing-library/react';

import PageHeader from '../PageHeader';

jest.mock('../../Navigation', () => () => (<nav>Navigation</nav>));
jest.mock('../../HeaderActions', () => () => (<nav>Header actions</nav>));
jest.mock('../../QuickSearch', () => () => (<form>Quick search</form>));
jest.mock('../../PageLogo', () => () => (<a>Page logo</a>));

test('PageHeader renders correctly', () => {

    const { container } = render(<PageHeader />);

    expect(container).toMatchSnapshot();
});

