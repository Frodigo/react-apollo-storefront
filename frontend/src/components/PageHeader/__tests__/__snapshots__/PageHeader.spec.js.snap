// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders correctly 1`] = `
<div>
  <header>
    <div
      class="d-none d-lg-block container"
    >
      <div
        class="d-flex align-items-center pb-3 pt-3 row"
      >
        <div
          class="col-sm-3"
        >
          <a>
            Page logo
          </a>
        </div>
        <div
          class="col-sm-6"
        >
          <form>
            Quick search
          </form>
        </div>
        <div
          class="col-sm-3"
        >
          <nav>
            Header actions
          </nav>
        </div>
      </div>
    </div>
    <nav>
      Navigation
    </nav>
  </header>
</div>
`;
